### PR TITLE
GH1775: Fix CakeScript tool resolution

### LINKED ISSUE #1775: Strange usage of Cake.Core.dll when executing cake sub process
When testing the Cake.Recipe Integration Tests (which spin up sub Cake processes to test building actual projects) I ran into this issue:

image

Where rather than executing Cake.exe, it was executing Cake.Core.dll.

### PR DESCRIPTION
**Pull request by devlead**
* Address Executing assembly issue
* Add support for Cake script on Cake.CoreCLR
* Remove no longer needed special .NET Core ToolPath in integration tests
* fixes #1775

### CODE REVIEW

File: src/Cake.Common/Tools/Cake/CakeRunner.cs [hunk 1]
```diff
@@ -150,7 +176,20 @@ protected override string GetToolName()
...
         protected override IEnumerable<string> GetToolExecutableNames()
         {
-            return new[] { "Cake.exe" };
+            return new[]
+            {
+#if NETCORE
+                "Cake.dll",
+                "cake",
+                "Cake",
+                "Cake.exe"
```
Comment (gep13) [inline src/Cake.Common/Tools/Cake/CakeRunner.cs]: Is this required here?

File: src/Cake.Common/Tools/Cake/CakeRunner.cs [hunk 2]
```diff
@@ -150,7 +176,20 @@ protected override string GetToolName()
...
+#else
+                "Cake.exe",
+                "cake",
+                "Cake",
+                "Cake.dll"
```
Comment (gep13) [inline src/Cake.Common/Tools/Cake/CakeRunner.cs]: Is this required here?

File: src/Cake.Common/Tools/Cake/CakeRunner.cs (hunk 1 above)
Comment (devlead) [reply src/Cake.Common/Tools/Cake/CakeRunner.cs]: well on windows if it finds a .exe it'll work.

File: src/Cake.Common/Tools/Cake/CakeRunner.cs (hunk 2 above)
Comment (devlead) [reply src/Cake.Common/Tools/Cake/CakeRunner.cs]: Well you could use Cake.CoreCLR  with full if no other found.

**Comment by devlead** [comment]
@gep13 rebased.

**Comment by devlead** [comment]
@cake-build/cake-team rebased.

**Comment by gep13** [comment]
@devlead your changes have been merged, thanks for your contribution 👍

**Pull request merged**